File: web/src/services/careerCopilot.ts
import api from "./api";

export interface CareerCopilotRequest {
  resume_data: Record<string, unknown>;
  target_role: string;
}

export async function runCareerCopilot(
  data: CareerCopilotRequest
) {
  const response = await api.post(
    "/career-copilot/run",
    data
  );

  return response.data;
}

File: backend/api/v1/career_copilot.py
from fastapi import APIRouter

from backend.schemas.career_copilot import CareerCopilotRequest
from backend.services.career_copilot_service import CareerCopilotService

router = APIRouter(
    prefix="/career-copilot",
    tags=["Career Copilot"],
)

service = CareerCopilotService()


@router.post("/run")
async def run(
    request: CareerCopilotRequest,
):

    return service.execute(
        request.resume_data,
        request.target_role,
    )